src/db/result_set.cr

```diff
@@ -99,6 +104,16 @@ module DB
       {% end %}
     end
 
+    private def internal_read(**types : **T) forall T
+      {% begin %}
+        NamedTuple.new(
+          {% for name, type in T %}
+            {{ name }}: read({{type.instance}}),
```

Comment:
This won't work in the general case. The query determines the order of the columns, and the code expects it to be the order to the fields in the tuple. The order in the named tuples is not guaranteed since it can be interchanged without affecting the semantic.

Reply:
Order of named tuples is guaranteed. It's only when you mix two named tuples with same names and types but different orders that they get unified to one of them.

Reply:
Great.
@bcardiff is there another problem ?
(about the naming, there is no link between the name of the column and the name in the tuple, it is redondant in the current state of the PR)

Reply:
Oh, but if you are passing a named tuple, then probably you should read values independent of the order, similar to how `DB.mapping` does it. You probably need to declare one variable for each key in the tuple, read them all, and construct the named tuple from all of that.

Reply:
@Nephos what @asterite mention in the above message is the concern I raised in the initial review message. You can to be sure that the `read` is reading the column named `name` otherwise the result will be wrong. `read` read the column in the order of the query and that does not match the order of the names of the named tuple members.

Reply:
It does not need to be matched. I might want to use a shorter name than the column: "article.title" => :title

Reply:
@Nephos The column has a fixed named, if `article.title` is used in sql query the name will be `title` AFAIK. I would expect that the name of the column should match exactly the name of the tuple entry.

Reply:
And I disagree, as I said above. It's very common that the names in the DB are bad, or unidiomatic for crystal, and you cannot change them.

Reply:
Maybe the best that `map_keys` can get is https://gist.github.com/bcardiff/b6c8fff62a5812e476a57e90c21a51c2 because you need some typing on that call, it needs to be a macro to be type safety.

Mapping keys it's something that would allow broader use cases for returning named tuples from db. I see them as independent topics.

@RX14 I'm sorry, but I don't get with what you disagree. Would you prefer a runtime error if the order of the query does not match the order of the keys in the named tuple?

Reply:
I'd prefer it to entirely disregard the keys and read off the values in order which is exactly what this pr does now and is clearly workable because non-named triples do it.

Reply:
@bcardiff is that ok for you ?
(The name of the key is not associated with the column's names at all)
If not, we should close this PR, and I'll eventually make a fork (and synchronize it with this repo) to add this feature.

Reply:
Perhaps we should have 2 names for the 2 behaviours.

Also, please can we move to traditional top-level comments from a line comment. They're more annoying and this change covers the whole scope of the PR.

Reply:
@Nephos yes, lets keep things simpler. The feature as it is is fine: the name of the key is not associated with the column's names at all. It's just a syntax sugar to avoid reading things in order and making a named tuple. 

If we could get some specs to cover the overloads that would be great.

Reply:
I'm on it

Reply:
Ok I've no idea where/how to make the specs :)

Reply:
Here are the specs for the other `query_one` overloads: https://github.com/crystal-lang/crystal-db/blob/39e17f82ca1ce15145f68f0c21e9d1b00cb2f21b/spec/dummy_driver_spec.cr#L112

Just copy one for tuples, adapt it for named tuples.

Reply:
Ok thanks. I think it's ok.